AlphaWallet/Market/Coordinators/UniversalLinkCoordinator.swift

```diff
@@ -147,10 +147,16 @@ class UniversalLinkCoordinator: Coordinator {
                     return
                 }
                 //filter null tokens
-                let filteredTokens = strongSelf.checkERC875TokensAreAvailable(
+                var filteredTokens = strongSelf.checkERC875TokensAreAvailable(
```

Comment:
We can change the `if` to an `if-else` and keep the `filteredTokens` as a `let` like this:

```
let filteredTokens: [String]
//if the token is spawnable, we don't check against the balance
if signedOrder.order.spawnable, let tokens = signedOrder.order.tokenIds {
    filteredTokens = [tokens[0].description]
} else {
    filteredTokens = strongSelf.checkERC875TokensAreAvailable(
            indices: signedOrder.order.indices,
            balance: balance
    )
}
```

I often break it myself, but I like this rule of thumb: an `if-else` is better than an `if` without `else` because it's always easy to miss out something if there's just a dangling `if` unless it contains an early return

But should this line:

```
filteredTokens = [tokens[0].description]
```

be this instead?

```
filteredTokens = tokens.map { $0.description }
```

And since the balance don't have to checked for spawnable tokens, we can re-organize the code like this:

```
if signedOrder.order.spawnable, let tokens = signedOrder.order.tokenIds {
    foo() //extracted method mentioned below
} else {
    //like it used to be, indented 1 more level now
    getERC875TokenBalanceCoordinator = GetERC875BalanceCoordinator(config: config)
    getERC875TokenBalanceCoordinator?.getERC875TokenBalance(for: recoverAddress, contract: contractAsAddress) { [weak self] result in
        //like it used to be, but extract the code starting from `if filteredTokens.isEmpty {` into a function foo() and call it
    }
}
```

So it'll actually be slightly faster if we redeem a spawnable link, skipping 1 RPC call